pandas/core/index.py

```diff
@@ -560,7 +566,18 @@ def _to_embed(self, keep_tz=False):
         return an array repr of this object, potentially casting to object
 
         """
-        return self.values
+        return self.values.copy()
+
+    def to_dense(self):
```

Comment:
Do we have "non-dense" index objects? This seems like a poor choice of name name and moreover largely unnecessary.

I suppose `to_ndarray` or `to_numpy_array` would be reasonable, but you might as well just use `np.asarray` directly.